android-agent/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -377,14 +379,30 @@ private void scheduleDiskTelemetryReader(
         }
     }
 
+    /**
+     * Adds a callback to be invoked after the OpenTelemetry SDK has been initialized. This can be
+     * used to defer some early lifecycle functionality until the working SDK is ready.
+     *
+     * @param callback - A callback that receives the OpenTelemetry SDK instance.
+     * @return this
+     */
+    public OpenTelemetryRumBuilder addOtelSdkReadyListener(Consumer<OpenTelemetrySdk> callback) {
```

Comment:
This reminds me of what I tried to do in this [other PR](https://github.com/open-telemetry/opentelemetry-android/pull/196), which so far seemed like it was superseded by the Instrumentation API, as it provides a way for instrumentations to get notified when the SDK is ready to be used. What use case is this method trying to solve?

Reply:
There's similarity for sure. It's a straightforward means for a component (in this case, the init events) to get notified immediately after the otel sdk has been initialized. In this case, the init events have to be buffered until they can be emitted. Without this register/notify mechanism, there's no way to know when it's "safe" to emit the events.

I made it a bit generic, because I thought that there could be other uses for this. However, there's the YAGNI school of thought, which might suggest that I remove this and just explicitly call a `flush` or `emitNow` method on the InitEvents instead. I don't feel strongly either way.